pennylane/pulse/parametrized_evolution.py

```diff
@@ -472,6 +473,52 @@ def hash(self):
             )
         )
 
+    def _flatten(self):
+
+        data = self.data
+        metadata = (tuple(self.t), self.H, self.hyperparameters, self.odeint_kwargs)
+
+        return data, metadata
+
+    @classmethod
+    def _unflatten(cls, data, metadata):
+        params = data
+        t, H, hyperparamameters, odeint_kwargs = metadata
+        return cls(
+            H,
+            params,
+            t,
+            complementary=hyperparamameters["complementary"],
+            return_intermediate=hyperparamameters["return_intermediate"],
+            **odeint_kwargs,
+        )
+
+    def __eq__(self, other):
+        """Check equality between two ParametrizedEvolution operators"""
+        if self.t is None:
+            same_times = other.t is None
+        else:
+            same_times = all(self.t == other.t)
+        if self.data is None:
+            same_data = other.data is None
+        elif isinstance(self.data, tuple) and isinstance(other.data, tuple):
+            same_data = self.data == other.data
+        elif isinstance(self.data, Iterable) and isinstance(other.data, Iterable):
```

Comment:
Under what circumstance does this happen? I thought we pass params to the `Operation` init and it always converts to a tuple. I think the tuple comparison in 504-505 should be sufficient for comparing `data` attributes.